apps/full-site-editing/full-site-editing-plugin/premium-content/subscription-service/class-wpcom-offline-subscription-service.php

```diff
@@ -0,0 +1,66 @@
+<?php declare( strict_types = 1 );
+/**
+ * @package A8C\FSE\Earn
+ *
+ * This subscription service is used when a subscriber is offline and a token is not available.
+ * This subscription service will be used when rendering content in email and reader on WPCOM only.
+ * When content is being rendered, the current user and site are set.
+ * This allows us to lookup a users subscriptions and determine if the
+ * offline visitor can view content that has been deemed "Premium content".
+ */
+namespace A8C\FSE\Earn\PremiumContent\SubscriptionService;
+
+// phpcs:ignore ImportDetection.Imports.RequireImports.Symbol
+class WPCOM_Offline_Subscription_Service extends WPCOM_Token_Subscription_Service {
+
+	/**
+	 * @return boolean
+	 */
+	public static function available() {
+		return parent::available() && is_user_logged_in();
```

Comment:
Just in case, I updated this condition to check if this request is from a running job (email) or from an API request (reader). I think that should avoid the token method being bypassed inadvertently.